keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1150 @@
+   Shareable *bool
+}
+
+// DeviceCapacity describes a quantity associated with a device.
+type DeviceCapacity struct {
+   // Value defines how much of a certain device capacity is available.
+   //
+   // +required
+   Value resource.Quantity
+
+   // SharingPolicy specifies that this device capacity must be consumed
+   // by each resource claim according to the defined sharing policy.
+   // The Device must be shareable.
+   //
+   // +optional
+   // +featureGate=DRAConsumableCapacity
+   SharingPolicy *CapacitySharingPolicy
+}
+
+// CapacitySharingPolicy defines how requests consume the available capacity.
+// The sharing policy can either specify a range of valid values or a discrete set of them.
+// Each policy has a default value, either explicitly or implicitly.
+// Exactly one of the consumption policies must be defined.
+type CapacitySharingPolicy struct {
+   // DiscreteValues defines a set of acceptable quantity values in consuming requests.
+   //
+   // +optional
+   // +oneOf=SharingPolicy
+   DiscreteValues *CapacitySharingPolicyDiscrete
+
+
+   // ValueRange defines an acceptable quantity value range in consuming requests.
+   //
+   // +optional
+   // +oneOf=SharingPolicy
+   ValueRange *CapacitySharingPolicyRange
+}
+
+// CapacitySharingPolicyDiscrete defines a set of discrete allowed capacity values.
+// If Options is not provided, only default value is valid.
+//
+// - If the requested amount is not listed in the options, it is rounded up to the next higher valid value.
+// - If the requested amount exceeds the maximum value in the available options, the request does not satisfy the policy,
+//   and the device cannot be allocated.
+type CapacitySharingPolicyDiscrete struct {
+    // Default specifies the default capacity to be used for a consumption request
+    // if no value is explicitly provided.
+    //
+    // +required
+    Default resource.Quantity
+
+    // Options defines a list of additional valid capacity values that can be requested.
+    // The Default must not be one of the Options.
+    //
+    // +optional
+    // +listType=atomic
+    Options []resource.Quantity
+}
+
+
+// CapacitySharingPolicyRange defines a valid range for consumable capacity values.
+//
+// - If the requested amount is less than Minimum, it is rounded up to the Minimum value.
+// - If the requested amount is between Minimum and Maximum, ChunkSize is set,
+//   and the amount does not align with the ChunkSize,
+//   it will be rounded up to the next value matching Minimum + (n * ChunkSize).
+// - If the requested or rounded amount exceeds Maximum (if set), the request does not satisfy the policy,
+//   and the device cannot be allocated.
+//
+// - If ChunkSize is not set, the requested amount is used as-is, provided it falls within the range.
+type CapacitySharingPolicyRange struct {
+    // Minimum specifies the minimum capacity allowed for a consumption request.
+    // This also acts as the default if no value is specified.
```

Comment:
is minimum always the sensible default? if so, why not choose the minimum of CapacitySharingPolicyDiscrete.Options implicitly as well? it seems unusual to give one an explicit default and the other an implicit